client/src/pages/ResourcePage.tsx

```diff
@@ -239,46 +69,114 @@ const ResourcePage = (): ReactElement => {
+
   return (
     <div className="resource-page-wrapper">
-      <Sidebar currentPage={pages.RESOURCES} />
+      <Sidebar currentPage={pages.RESOURCES}></Sidebar>
       <Header />
       <img className="page-svg" alt="Resource Page" src={ResourcePageSVG} />
       <div className="resource-page-content">
-        <div className="page-title">Resource Hub</div>
+        <div className="resource-title-container">
+          <div className="resource-title-item">Resource Hub</div>
+          <div className="resource-title-item">
+            <div className="resource-toggle-group">
+              <Button
+                className={'toggle-button' && (!edit ? 'active' : '')}
+                onClick={cancelEdit}
+              >
+                View Resources
+              </Button>
+              <Button
+                className={'toggle-button' && (edit ? 'active' : '')}
+                onClick={enableEdit}
+              >
+                Edit Resources
+              </Button>
+            </div>
+          </div>
+        </div>
+
         <div className="resource-toggle-group">
-          {resourceRoleBtns.map((button) => (
-            <Button
-              key={button.value}
-              className={`toggle-button ${
-                button.value === currentValue && 'active'
-              }`}
-              onClick={() =>
-                handleResourceChange(button.resources, button.value)
-              }
-            >
-              {button.value}
-            </Button>
-          ))}
+          {!edit ? (
+            Object.keys(resourcesPerRole).map((role) => (
+              <Button
+                key={role}
+                className={`toggle-button ${role === currentValue && 'active'}`}
+                onClick={() => handleResourceChange(role)}
+              >
+                {role}
+              </Button>
+            ))
+          ) : (
+            <AddResourceModal onAdd={filterResources} />
+          )}
         </div>
+
         <div className="resource-btn-group">
-          {resources.map((button) => (
-            <Button
-              onClick={() => handleResourceClick(button.link)}
-              className="resource-btn"
-              key={button.key}
-            >
-              {button.buttonText}
-            </Button>
-          ))}
+          {!edit
+            ? resourcesPerRole[currentValue].map((resource, idx) => (
+                <Button
+                  onClick={() => handleResourceClick(resource.link)}
+                  className="resource-btn"
+                  key={idx}
+                >
+                  {resource.name}
+                </Button>
+              ))
+            : resourcesPerRole[currentValue].map((resource, idx) => (
+                <div key={idx} className="editable-resource">
+                  <Button
+                    className="resource-btn"
+                    key={idx}
+                    onClick={() => {
+                      setShowEditModal(true);
```

Comment:
After submitting an edit for a resource, the EditResourceModal doesn't show up when I click on the other resources but I'm not sure why it's happening.

Reply:
Got it, I wasn't calling the callback function on submit. Thanks!!